cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -2529,7 +2533,197 @@ def union_dependent_cleaning(df):
+            "No",
+            [0, "No", None, ["No", None], 0, None, "Yes", "Negative"],
+        ],
+        [*think_had_covid_cols, "other_covid_infection_test", "other_covid_infection_test_results"],
+    )
+    # 5
+    df = assign_column_value_from_multiple_column_map(
+        df,
+        "think_had_covid",
+        ["Yes", ["No", "Yes", "Positive"]],
+        ["think_had_covid", "other_covid_infection_test", "other_covid_infection_test_results"],
+    )
+    # 6
+    for col in covid_test_cols:
+        df = df.withColumn(
+            col,
+            F.when(
+                (all_columns_values_in_list(covid_test_cols, ["No", "Negative", None]))
+                & (F.col("think_had_covid") == "Yes")
+                & (F.col("think_had_covid_onset_date").isNull()),
+                None,
+            ).otherwise(F.col(col)),
+        )
+    # 7
+    df = assign_column_value_from_multiple_column_map(
+        df,
+        "other_covid_infection_test",
+        [None, ["No", "No", None, None, None, 0, 0]],
+        [
+            "other_covid_infection_test",
+            "think_had_covid",
+            "think_had_covid_contacted_nhs",
+            "think_had_covid_admitted_to_hospital",
+            "other_covid_infection_test_results",
+            "sympt_covid_count",
+            "survey_response_dataset_major_version",
+        ],
+    )
+    # 8
+    for col in hospital_covid_cols:
+        df = df.withColumn(
+            col,
+            F.when(
+                (F.col(col) == "No")
+                & (F.col("think_had_covid") == "No")
+                & (F.col("think_had_covid_onset_date").isNull())
+                & (F.col("sympt_covid_count") == 0)
+                & (F.col("survey_response_dataset_major_version") == 1)
+                & (
+                    reduce(
+                        add,
+                        [F.when(F.col(c).isin(["No", "Negative", None]), 1).otherwise(0) for c in hospital_covid_cols],
+                    )
+                    == 1
+                ),
+                None,
+            ).otherwise(F.col(col)),
+        )
+    # 9
+    count_no = reduce(
+        add,
+        [
+            *[F.when(F.col(c).eqNullsafe("No"), 1).otherwise(0) for c in hospital_covid_cols],
+            F.when(
+                F.col("other_covid_infection_test") == "No", F.col("survey_response_dataset_major_version")
+            ).otherwise(0),
+        ],
+    )
+    count_yes = count_occurrence_in_row([*hospital_covid_cols, "other_covid_infection_test"], "Yes")
+    for col in [*hospital_covid_cols, "other_covid_infection_test"]:
+        df = df.withColumn(
+            col,
+            F.when(
+                (F.col(col).eqNullSafe("No"))
+                & (F.col("think_had_covid").isNull())
+                & (count_no <= 2)
+                & (F.col("other_covid_infection_test_results").isNull())
+                & (F.col("survey_response_dataset_major_version") == 0)
+            ),
+        )
```

Comment:
This is #10 (correcting for my sanity when checking against STATA logic on confluence)